azurerm/resource_arm_servicebus_namespace_authorization_rule.go

```diff
@@ -0,0 +1,217 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/set"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmServiceBusNamespaceAuthorizationRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmServiceBusNamespaceAuthorizationRuleCreateUpdate,
+		Read:   resourceArmServiceBusNamespaceAuthorizationRuleRead,
+		Update: resourceArmServiceBusNamespaceAuthorizationRuleCreateUpdate,
+		Delete: resourceArmServiceBusNamespaceAuthorizationRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"namespace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateServiceBusNamespaceName(),
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"rights": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(servicebus.Listen),
+						string(servicebus.Send),
+						string(servicebus.Manage),
+					}, true),
+				},
+				Set: set.HashStringIgnoreCase,
```

Comment:
can we switch this over to using Bools rather than a Set, since this makes the diff clearer (and then these can be inferred)?